vault/resource_mongodbatlas_secret_backend.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"golang.org/x/net/context"
+)
+
+var mongodbAtlasAPIFields = []string{
+	consts.FieldPublicKey,
+	consts.FieldPrivateKey,
+}
+
+func mongodbAtlasSecretBackendResource() *schema.Resource {
+	return provider.MustAddMountMigrationSchema(&schema.Resource{
```

Comment:
So after having a closer look at the code, I believe we can remove this. The `vault_mount` resource is responsible for remounting/unmounting, and if there is a path change it will always perform mount migration.
```suggestion
	&schema.Resource{
```